packages/driver/cypress/integration/commands/request_spec.js

```diff
@@ -473,6 +473,34 @@ describe('src/cy/commands/request', () => {
       })
     })
 
+    describe('binary data', () => {
+      // https://github.com/cypress-io/cypress/issues/6178
+      it('can send Blob', () => {
+        const body = new Blob([[1, 2, 3, 4]], { type: 'application/octet-stream' })
+
+        cy.request(
+          {
+            body,
+            method: 'POST',
+            url: 'http://localhost:3500/dump-octet-body',
+            headers: {
+              'Content-Type': 'application/octet-stream',
+            },
+          },
+        )
+        .then((response) => {
+          expect(response.status).to.equal(200)
+
+          // When user-passed body to the Nodejs server is a Buffer,
+          // Nodejs doesn't provide any decoder in the response.
+          // So, we need to decode it ourselves.
+          const dec = new TextDecoder()
+
+          expect(dec.decode(response.body)).to.contain('1,2,3,4')
```

Comment:
I thought adding this decoding part to Cypress server because `content-type: text/html` with `ArrayBuffer` body is a bit weird. But I also thought that it's how Express.js works and we need to respect that. 

I didn't experiment how other frameworks work.